### PR TITLE
Pytest follow-up

### PR DESCRIPTION
**Pull request by djarecka**
This is a follow-up to my previous Pytes PR (#1722), this time I've read the Satra review :) 
* removed some files from tools directory that are not used anymore
* removed fixes
* changed .coveragerc (workflows is not omitted) 
* removed coverage from travis, so it is now calculated from CircleCI only 
* added one auto tests `test_auto_ACompCor.py` and changing `test_auto_TCompCor.py`, so it agrees with checkspec.py output
* recovered `test_BEDPOSTX.py` and `test_Xfibres.py`
* removed nosetes from `__init__.py` 
* changed one test from `test_math.py` that was failing
* some smaller changes suggested in the review 

Still TODO:
* tests `test_BEDPOSTX.py` and `test_Xfibres.py` don't affect anything IMO. Should I create `test_BEDPOSTX5.py` and `test_Xfibres5.py`, and delete  `test_auto_BEDPOSTX5.py` and `test_auto_Xfibres5.py`?
* `__init__.py` is not finished, `nipype.test()` doesn't work

### CODE REVIEW

File: nipype/interfaces/fsl/tests/test_maths.py [hunk 1]
```diff
@@ -207,10 +207,8 @@ def test_stdimage(create_files_in_directory):
         assert stder.cmdline == cmdline%dim
 
     # Test the auto naming
-    stder = fsl.StdImage(in_file="a.nii")
-    #NOTE_dj, FAIL: this is failing (even the original version of the test with pytest)
-    #NOTE_dj: not sure if this should pass, it uses cmdline from interface.base.CommandLine
-    #assert stder.cmdline == "fslmaths a.nii -Tstd %s"%os.path.join(testdir, "a_std.nii")
+    stder = fsl.StdImage(in_file="a.nii", output_type='NIFTI')
+    assert stder.cmdline == "fslmaths a.nii -Tstd %s"%os.path.join(testdir, "a_std.nii")
```
Comment (satra) [inline nipype/interfaces/fsl/tests/test_maths.py]: use `.format()`

**Comment by satra** [comment]
@djarecka - `test_bedpostx` and `test_xfibres` simply prevent the autocreation of the auto tests for those which would duplicate the `test_auto_*5` at a point in the past we had two versions of these tools. so your current changes are fine and it doesn't need `test_*5.py`

**Comment by satra** [comment]
@djarecka - if you can finish the `nipype.test()` this should be ready. i'll also send a PR to your branch with some updated other tests.

**Comment by djarecka** [comment]
@satra even without `test_bedpostx` and `test_xfibres` auto tests are NOT created.

**Comment by djarecka** [comment]
@satra I'm a bit stuck with `nipype.test()`. If you want to merge this ASAP, it might be better if I move `nipype.test()` to another PR, it somehow independent issue.

**Comment by satra** [comment]
@djarecka - you are correct - the autotests are not created because they are the same exact class as the `*5.py` classes in this case. so yes, in this case those two tests can be deleted.

**Comment by satra** [comment]
`test_BEDPOSTX.py` and `test_Xfibres.py`

**Comment by djarecka** [comment]
@satra ok, I've removed the extra tests. Just wasn't sure, if you want to disable auto tests for `BEDPOSTX5` and `XFibres5`.

I also changed earlier `%s` to `format()`  in `test_math` in the places where I could do it without changing the code itself (you're using `%s` style in `nipype.interfaces.fsl.maths`)

**Comment by djarecka** [comment]
@satra can you also check this assert? https://github.com/nipy/nipype/blob/master/nipype/interfaces/tests/test_base.py#L137
I've removed it since I'm only getting a UserWarning

**Comment by satra** [comment]
@djarecka -  i can't seem to figure out why that should raise a `TypeError` - ok to remove for now.

**Comment by satra** [comment]
regarding `.format` to slowly start moving towards current python standards - so a new change should always reflect current coding standards.

**Pull request merged**